lib/data/data_source/remote/authentication_remote_source_impl.dart

```diff
@@ -8,16 +9,28 @@ import '../../../domain/utils/success.dart';
 import '../../dto/user/create_user_dto.dart';
 import '../../dto/user/login_dto.dart';
 import '../../dto/user/update_user_dto.dart';
+import '../../dto/user/user_profile_dto.dart';
 
 @Injectable(as: AuthenticationRemoteSource)
 class AuthenticationRemoteSourceImpl implements AuthenticationRemoteSource {
+  AuthenticationRemoteSourceImpl(this.firestore, this.firebaseAuth);
+
+  final FirebaseFirestore firestore;
+  final FirebaseAuth firebaseAuth;
+
   @override
   Future<UserCredential> createUser(CreateUserDto dto) async {
     try {
-      final user = await FirebaseAuth.instance.createUserWithEmailAndPassword(
+      final user = await firebaseAuth.createUserWithEmailAndPassword(
         email: dto.email,
         password: dto.password,
       );
+      if (user.user != null) {
+        final userId = user.user!.uid;
+        await firestore.collection("users").doc(userId).set(
+              UserProfileDto(name: '', userId: user.user!.uid, initialLanguage: 'pl', userFolders: []).toJson(),
+            );
+      }
```

Comment:
I don't think the logic of updating user should be here, there is a method called `updateUser` where user update his username on Firebase. I think you should modify this method to accept the required params and create the users collection